tools/elf2bin.py

```diff
@@ -55,28 +55,29 @@ def get_segment_size_addr(elf, segment, path):
     raise Exception('Unable to find size and start point in file "' + elf + '" for "' + segment + '"')
 
 def read_segment(elf, segment, path):
-    tmpfile, dumpfile = tempfile.mkstemp()
-    os.close(tmpfile)
-    p = subprocess.check_call([path + "/xtensa-lx106-elf-objcopy", '-O', 'binary', '--only-section=' + segment, elf, dumpfile], stdout=subprocess.PIPE)
-    binfile = open(dumpfile, "rb")
-    raw = binfile.read()
-    binfile.close()
+    fd, tmpfile = tempfile.mkstemp()
+    os.close(fd)
+    p = subprocess.check_call([path + "/xtensa-lx106-elf-objcopy", '-O', 'binary', '--only-section=' + segment, elf, tmpfile], stdout=subprocess.PIPE)
+    with open(tmpfile, "rb") as f:
+        raw = f.read()
+    os.remove(tmpfile)
```

Comment:
re TemporaryDirectory, just removing the file right here also works.

Reply:
Nice and simple!